JBPM-8067: Stunner - Magnets & connector bendpoints are not working with the growable canvas.

romartin opened the pull request:
Hey @hasys  @tiagodolphine 

See https://issues.jboss.org/browse/JBPM-8067

Can you review please?

It depends on
https://github.com/kiegroup/lienzo-core/pull/79
https://github.com/kiegroup/lienzo-tests/pull/57

Thanks!

romartin commented:
Jenkins execute full downstream build

romartin commented:
Good point @tiagodolphine , sorry for the bug. I also found another one - when doing multiple selection and dragging it (so dragging the rectangle area that appears), after drag complete the CPs for the selected connector/s are being hidden.
So working on these two bugs now and will let you know once ready again.
Thanks!

romartin commented:
Hey @tiagodolphine  @hasys 

I've fixed the bug about not being able to click (select) a connector. Rather than fixing it I'd been trying to improve the usability in general again. Now I think it works fine and the user experience is good. 

Once moving over a connector, it displays the existing connector's control points and also displays the location of the mouse pointer, that's similar as it was before, but now the key point is when performing the "mouse down" over the connector - once doing it, it displays an arc that is being animated while keeping the mouse down, and once the arc draws a complete circle, a new control point is being created and so you can just start moving the mouse (dragging) for changing its location. Otherwise, if you just do a regular click on the connector or performs a "mouse up" before the arc is complete, it just selects it.

Also I've fixed some other bug I found related to connectors and control points and updated/added unit tests. So ready for review again.

Thanks!

romartin commented:
Jenkins execute full downstream build

romartin commented:
Jenkins execute full downstream build

hasys commented:
Hi @romartin looks, good but I am still not able to:
* Reconnect bendpoint from one node to another if nodes placed out of the canvas. I am able to disconnect the point from the node, but not able to connect it again/reconnect to another point.
* Add new bendpoint if sequence flow is out of the canvas. New nice animation shows successfully, but new bendpoint is not created.

Will check the code now.

hasys commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/AbstractCanvas.java:
```diff
@@ -202,9 +192,18 @@ public Canvas deleteShape(final Shape shape) {
+        if (shape.getUUID() == null) {
+            shape.setUUID(UUID.uuid());
+        }
+        shape.getShapeView().setUUID(shape.getUUID());
```
Shouldn't it be a part of `shape#setUUID` method?

romartin replied:
fixed,  thanks!

romartin commented:
Hey @hasys 

Thanks for the review! Good catches! :+1: 

I've rebased and fixed the comments on the code. About the issues when moving things "out of" the canvas - yeahh good points, I agree. But I would prefer to do this in a separate task, so just opened and start working on [JBPM-8111](https://issues.jboss.org/browse/JBPM-8111), trying this way to split tasks and tests for better reviews, is that good for you? If so, can I proceed on merging this??

Thanks!!

romartin commented:
Jenkins execute full downstream build

romartin commented:
Jenkins please retest this

romartin commented:
Jenkins execute full downstream build

romartin commented:
rebased and squashed

romartin commented:
Jenkins please retest this
